tensorboard/plugins/profile/tf_op_profile/tf-op-table.html

```diff
@@ -0,0 +1,209 @@
+</div>
+<template is="dom-if" if="[[expanded]]">
+  <template is="dom-repeat" items="{{node.children}}" sort="_sortEntry">
+    <tf-op-table-entry
+      node="[[item]]"
+      level={{_nextLevel(level)}}
+      table={{table}}>
+    </tf-op-table-entry>
+  </template>
+</template>
+</dom-module>
+
+<script>
+import {flameColor, percent, utilization} from './utils.js';
+
+Polymer({
+  is: 'tf-op-table-entry',
+  properties: {
+    node: {
+      type: Object,
+    },
+    level: {
+      type: Number,
+      value: 0,
+    },
+    table: {
+      type: Object,
+    },
+    expanded: {
+      type: Boolean,
+      value: false,
+      notify: true,
+    },
+  },
+  _eq: function(x, y) {
+    return x == y;
+  },
+  _sortEntry: function(left, right) {
+    var leftTime = left.metrics ? left.metrics.time : 0;
+    var rightTime = right.metrics ? right.metrics.time : 0;
+    return rightTime - leftTime;
+  },
+  ready: function() {
+    if (!this.node || !this.node.metrics || !this.$.bar) return;
+    this.$.bar.style.width = percent(this.node.metrics.time);
+    this.$.bar.style.backgroundColor = flameColor(utilization(this.node), 1, 0.2);
+  },
+  _nextLevel: function(level) { return level + 1; },
+  _handleHeaderClick: function(e) {
+    this.expanded ^= true;
+    if (this.table._selectedEntry) this.table._selectedEntry.classList.remove('selected');
+    this.classList.add('selected');
+    this.table._selectedEntry = this;
+    this.table.selected = this.node;
+    console.log("Selected ", this.node);
+  },
+  _handleHeaderMouseEnter: function(e) { this.table.active = this.node; },
+  _handleHeaderMouseLeave: function(e) { this.table.active = this.table.selected; },
+  _percent: percent,
+  _provenance: function(node) {
+    return (node.xla && node.xla.provenance)
+              ? node.xla.provenance.replace(/^.*\//, '') : "";
+  },
+});
+</script>
+
+<!--
+  A tf-op-table shows a hierarchical profile using a tree of tf-op-table-entrys.
+  One entry at a time can be made active (by clicking on it).
+-->
+<dom-module id="tf-op-table">
+<template>
+<style include="tf-op-table-styles">
+:host {
+  display: block;
+  background-color: white;
+}
+#header {
+  font-size: smaller;
+  font-weight: bold;
+  color: #666;
+  padding-bottom: 0.25em;
+  border-bottom: 1px solid #666;
+  text-transform: uppercase;
+}
+/* Match indented names */
+#name { padding-left: 2em; }
+</style>
+<div id="header">
+  <span id="time">Time</span>
+  <span id="name">Name</span>
+  <span id="provenance">TensorFlow Op</span>
+</div>
+<tf-op-table-entry node="[[rootNode]]" table="[[_this()]]" expanded="true">
+</tf-op-table-entry>
+</dom-module>
+
+<script>
+Polymer({
+  is: 'tf-op-table',
+  properties: {
+    rootNode: {
+      type: Object,
+    },
+    active: {
+      type: Object,
+      notify: true,
+    },
+    selected: {
+      type: Object,
+      value: null,
+    },
+  },
+  _this: function() { return this; },
```

Comment:
It'd probably be neater to bind individual properties instead of pass in the containing component. In terms of encapsulation of logic, the entries do not need to know all of the properties of the table.